link/internal/services/machine.go

```diff
@@ -0,0 +1,95 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	logger "linkSM/internal/utils"
+	"log/slog"
+)
+
+type createMachineParams struct {
+	Name             string `json:"name" `
+	Isrepairrequired bool   `json:"Isrepairrequired",omitempty`
+	Isactive         bool   `json:"Isactive",omitempty`
+}
+
+func CreateMachine(data []byte, log *slog.Logger, url string) ([]byte, error) {
+	log.Info("Start processing shift creation request")
+
+	// map[string]interface{}, because the request may not match the structure
+	var rawData map[string]interface{}
```

Comment:
it will be better to validate it firstly if it is not match the structure - return error!  Don't overcomplicate things) (controller should be responsible for validation)